File: simple.py
from nevow import athena, loaders

class Button(athena.LiveFragment):
    jsClass = u"SimpleWidget.Button"
    docFactory = loaders.xmlstr("""
<div xmlns:nevow="http://nevow.com/ns/nevow/0.1"
     nevow:render="liveFragment">
  <form onSubmit="return false;">
    <input type="submit" value="" onClick="SimpleWidget.Button.get(this).click()"/>
  </form>
</div>""")
    allowedMethods = ['onClick', 'getLabel']
    def __init__(self, label):
        athena.LiveFragment.__init__(self)
        self.label = label

    def getLabel(self):
        return unicode(self.label)
    
    def onClick(self):
        print "clicked %s" % self.label

File: simple.js
// import Nevow.Athena

if (typeof SimpleWidget == 'undefined') {
    SimpleWidget = {};
}

SimpleWidget.Button = Nevow.Athena.Widget.subclass();
SimpleWidget.Button.methods(
  function __init__(self, node) {
    SimpleWidget.Button.upcall(self, '__init__', node);
    self.callRemote('getLabel').addCallback(function (label) {
	input = node.getElementsByTagName("INPUT")[0];
	input.value = label;
    });
  },
  function click(self) {
      self.callRemote('onClick').addErrback(
        function (err) { Divmod.err(err); });
  });
